packages/mui-codemod/src/util/findComponentJSX.js

```diff
@@ -17,8 +17,10 @@ export default function findComponentJSX(j, options, callback) {
 
   root
     .find(j.ImportDeclaration)
-    .filter((path) =>
-      path.node.source.value.match(new RegExp(`^@mui/material/?(${componentName})?`)),
+    .filter(
+      (path) =>
+        path.node.source.value.match(new RegExp(`^@mui/material/(${componentName})`)) ||
+        path.node.source.value === '@mui/material',
```

Comment:
We can make it a little simpler:

```suggestion
    .filter((path) =>
      path.node.source.value.match(new RegExp(`^@mui/material(/${componentName})?$`)),
```

It's the same as your fix, as `$` indicates to match the end of the line, so it's either

- `@mui/material` and end of line, so nothing else
- `@mui/material/componentName` and end of line, so nothing else

Reply:
Thanks for clear explaination. fixed in this commit https://github.com/mui/material-ui/pull/40855/commits/8b714f8cb786a6ac906c9dc783b38c0e86979afd

Reply:
Nice! Did you test if it's fixed on your side after the refactor?

Reply:
Yes, ran codemod script.  Now util is filtering out data correctly